modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/rules/RuleEditorModal.js

```diff
@@ -0,0 +1,198 @@
+	);
+
+	/**
+	 * This reference is used for updating rule value without causing a re-render
+	 */
+	const ruleRef = useRef(rule);
+
+	const {resource: rolesResource} = useResource({
+		fetch,
+		link: `${window.location.origin}/o/headless-admin-user/v1.0/roles`,
+	});
+
+	const roles = useMemo(
+		() =>
+			rolesResource?.items.map(({id, name}) => ({
+				id: `${id}`,
+				label: name,
+				name,
+				value: name,
+			})),
+		[rolesResource]
+	);
+
+	const transformedPages = useMemo(() => getTransformedPages(pages), [pages]);
+	const fields = useMemo(() => getFields(pages), [pages]);
+
+	return (
+		<>
+			<ClayModal.Header>
+				{rule
+					? Liferay.Language.get('edit-rule')
+					: Liferay.Language.get('create-new-rule')}
+			</ClayModal.Header>
+			<ClayModal.Header withTitle={false}>
+				<ClayInput.Group className="pl-4 pr-4">
+					<ClayInput.GroupItem>
+						<ClayInput
+							aria-label={Liferay.Language.get('rule-title')}
+							className="form-control-inline"
+							onChange={({target: {value}}) => setRuleName(value)}
+							placeholder={Liferay.Language.get('untitled-rule')}
+							type="text"
+							value={ruleName}
+						/>
+					</ClayInput.GroupItem>
+				</ClayInput.Group>
+			</ClayModal.Header>
+			<ClayModal.Body>
+				<Editor
+					allowActions={[
+						'show',
+						'enable',
+						'require',
+						'calculate',
+						'jump-to-page',
+					]}
+					className="pl-4 pr-4"
+					dataProvider={[]}
+					fields={fields}
+					functionsURL={functionsURL}
+					onChange={(rule) => {
+						ruleRef.current = rule;
+					}}
+					onValidator={(value) => setInvalidRule(!value)}
+					operatorsByType={functionsMetadata}
+					pages={transformedPages}
+					roles={roles}
+					rule={rule}
+				/>
+			</ClayModal.Body>
+			<ClayModal.Footer
+				last={
+					<ClayButton.Group spaced>
+						<ClayButton displayType="secondary" onClick={onClose}>
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+						<ClayButton
+							disabled={invalidRule || !ruleName}
+							onClick={() => {
+								onClick({
+									dataRule: {
+										...ruleRef.current,
+										name: ruleName,
+									},
+									loc: rule?.ruleEditedIndex,
+									rule,
+								});
+								onClose();
+							}}
+						>
+							{Liferay.Language.get('save')}
+						</ClayButton>
+					</ClayButton.Group>
+				}
+			/>
+		</>
+	);
+};
+
+export default ({
+	functionsMetadata,
+	functionsURL,
+	isVisible,
+	onClick,
+	onClose: onCloseFn,
+	rule,
+}) => {
+	const {pages} = useFormState();
+	const {observer, onClose} = useModal({
+		onClose: onCloseFn,
+	});
+
+	if (!isVisible) {
+		return null;
+	}
+
+	return (
+		<ClayModal
+			className="data-layout-builder-editor-modal"
+			observer={observer}
+			size="full-screen"
+		>
+			<RuleEditorModalContent
+				functionsMetadata={functionsMetadata}
+				functionsURL={functionsURL}
+				onClick={onClick}
+				onClose={onClose}
+				pages={pages}
+				rule={rule}
+			/>
+		</ClayModal>
+	);
+};
```

Comment:
The new base uses `ClayModalProvider`, so you can just create `RuleEditorModalBody` for example, and call directly from `RulesSidebar`.

See how it is done in the Editor: https://github.com/liferay/liferay-portal/blob/master/modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/Conditions.es.js#L266-L312

Reply:
I remember that when we implement rules in DE, we need to use it as it is today because the API does not allow adding 2 headers, take a look at this implementation:

https://github.com/liferay-forms/liferay-portal/pull/183/files/66de69d86daaa76aa7dc47905d21425892112579..2284ea3e5f91052a8d15f1e0de6fb9df987b4455#diff-de824e1a95ec00fe6ae39cb5b2aa0dfc518b323256608ff925d019f36e02500dR94-R112

Still, is it possible to use the API?

Reply:
Oh no 💆‍♂️!!! Well, in fact, it is a limiter, we can collaborate with Clay to think of ways to support something like that that could allow us to do this without having to add two headers. I will remember to open an issue to study this later.

Reply:
Sounds good, Matu! Thanks